### PR TITLE
FEATURE: Allow configuring static factory methods in Objects.yaml

### PR DESCRIPTION
**Pull request by albe**
This change allows to configure static factory methods in `Objects.yaml` by only specifying a `factoryMethodName` and leaving out `factoryObjectName`.

Example:
```
Acme\My\Class:
  factoryMethodName: Acme\My\Class::fromStatic
  arguments:
    1:
      setting: Acme.My.Class.ConfigurableValue
```

Before this would have required to create a non-static factory method inside a dedicated factory class (to avoid cyclic instanciation).

### CODE REVIEW

File: Neos.Utility.ObjectHandling/Classes/ObjectAccess.php [hunk 1]
```diff
@@ -435,45 +435,13 @@ public static function buildSetterMethodName(string $propertyName): string
...
-            case 6:
-                $object = new $className($arguments[0], $arguments[1], $arguments[2], $arguments[3], $arguments[4], $arguments[5]);
-            break;
-            default:
-                $class = new \ReflectionClass($className);
-                $object = $class->newInstanceArgs($arguments);
-        }
-
-        return $object;
+        return new $className(...$arguments);
```
Comment (albe) [inline Neos.Utility.ObjectHandling/Classes/ObjectAccess.php:445]: Not directly related to the change at hand, but why didn't we do this earlier? It's also faster https://gist.github.com/nikic/6390366

File: Neos.Flow/Classes/ObjectManagement/ObjectManager.php [hunk 2]
```diff
@@ -538,7 +538,7 @@ protected function instantiateClass($className, array $arguments)
         }
 
         try {
-            $object = ObjectAccess::instantiateClass($className, $arguments);
+            $object = new $className(...$arguments);
```
Comment (albe) [inline Neos.Flow/Classes/ObjectManagement/ObjectManager.php:541]: See below...

File: Neos.Flow/Classes/ObjectManagement/Configuration/Configuration.php [hunk 3]
```diff
@@ -191,6 +191,9 @@ public function getPackageKey()
     public function setFactoryObjectName($objectName)
     {
         $this->factoryObjectName = $objectName;
+        if ($this->factoryMethodName === '') {
+            $this->factoryMethodName = 'create';
```
Comment (bwaidelich) [inline Neos.Flow/Classes/ObjectManagement/Configuration/Configuration.php:195]: This seems weird, why is this change required?

File: Neos.Utility.ObjectHandling/Classes/ObjectAccess.php (hunk 1 above)
Comment (bwaidelich) [reply Neos.Utility.ObjectHandling/Classes/ObjectAccess.php:445]: I think this change makes a lot of sense, but I would prefer a separate PR for this

**Comment by bwaidelich** [comment]
Also we have to keep in mind to adjust #1964 once this is merged

File: Neos.Flow/Classes/ObjectManagement/Configuration/Configuration.php (hunk 3 above)
Comment (albe) [reply Neos.Flow/Classes/ObjectManagement/Configuration/Configuration.php:195]: Because `factoryMethodName` was sets with a default value in the property declaration, which led to ALL classes being handled as if they should be created by factory.
`Error: Method "create" does not exist.`

**Comment by albe** [comment]
Ah, keeping me busy by being pedantic on the details :D ❤️ You are totally right though. Will extract the optimization (which should actually end up in 5.3+ tbf) and add some docs and a test.

File: Neos.Flow/Classes/ObjectManagement/Configuration/Configuration.php (hunk 3 above)
Comment (albe) [inline Neos.Flow/Classes/ObjectManagement/Configuration/Configuration.php:195]:
```suggestion
            // Needed for b/c because all configured factory objects should default to 'create' method, but not having a factory object should not lead to a global static 'create' factory method
            $this->factoryMethodName = 'create';
```

File: Neos.Utility.ObjectHandling/Classes/ObjectAccess.php (hunk 1 above)
Comment (albe) [reply Neos.Utility.ObjectHandling/Classes/ObjectAccess.php:445]: See #1972

File: Neos.Flow/Classes/ObjectManagement/Configuration/Configuration.php (hunk 3 above)
Comment (albe) [reply Neos.Flow/Classes/ObjectManagement/Configuration/Configuration.php:195]: Would this help explain this odd line of code @bwaidelich ?

Comment (bwaidelich) [reply Neos.Flow/Classes/ObjectManagement/Configuration/Configuration.php:195]: Yes, thanks! :)

**Comment by bwaidelich** [comment]
> Ah, keeping me busy by being pedantic on the details :D

I'll just say "Edge-cases be edgy" 😛

**Comment by albe** [comment]
Okay, that should be it @bwaidelich
